mlflow/entities/trace_info.py

```diff
@@ -21,34 +21,13 @@ class TraceInfo(_MLflowObject):
         tags: tags associated with the trace.
     """
 
-    def __init__(
-        self,
-        request_id: str,
-        experiment_id: str,
-        timestamp_ms: int,
-        execution_time_ms: int,
-        status: TraceStatus,
-        request_metadata: Optional[List[TraceRequestMetadata]] = None,
-        tags: Optional[List[TraceTag]] = None,
-    ):
-        if request_id is None:
-            raise MlflowException("`request_id` cannot be None.")
-        if experiment_id is None:
-            raise MlflowException("`experiment_id` cannot be None.")
-        if timestamp_ms is None:
-            raise MlflowException("`timestamp_ms` cannot be None.")
-        if execution_time_ms is None:
-            raise MlflowException("`execution_time_ms` cannot be None.")
-        if status is None:
-            raise MlflowException("`status` cannot be None.")
-
-        self.request_id = request_id
-        self.experiment_id = experiment_id
-        self.timestamp_ms = timestamp_ms
-        self.execution_time_ms = execution_time_ms
-        self.status = status
-        self.request_metadata = request_metadata
-        self.tags = tags
+    request_id: str
+    experiment_id: str
+    timestamp_ms: int
+    execution_time_ms: Optional[int]
```

Comment:
Note: If we follow the same principle to span, `execution_time_ms` can be null if the trace doesn't finish properly.

Reply:
wouldn't we set the time within TTL logic to record the point at which the TraceStatus is marked as `ERROR` and the trace is closed for export of the failed state?

Reply:
Yeah on second thoughts, I agree we should set it to the time when trace is closed. But we still need to leave this as Optional, because `TraceInfo` object is instantiated when a trace starts (cuz we need to hold sone info like tags that are set at creation time). At this moment we can only set `None`, which it's possible to guarantee non-null at the time of closing.

Reply:
Excellent point! Agreed :)